osu.Game/Rulesets/Mods/Mod.cs

```diff
@@ -66,6 +66,12 @@ public abstract class Mod : IMod, IJsonSerializable
         [JsonIgnore]
         public virtual Type[] IncompatibleMods => new Type[] { };
 
+        /// <summary>
+        /// Returns whether this mod is a ruleset system mod and cannot be user-selected
+        /// </summary>         
+        [JsonIgnore]
+        public virtual bool System => false;
```

Comment:
I think this could be unnecessary as it feels redundant. Adding `ModType.System` and having `OsuModTouchDevice` set said type should be enough.
If `GetAllMods()` returns all then you can filter or retrieve them directly via `GetModsFor(ModType.System)`.

Reply:
This actually sounds pretty good. Forgot about the `ModType` grouping.

@smoogipoo sound good?